src/io/flutter/perf/EditorPerfDecorations.java

```diff
@@ -0,0 +1,502 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.perf;
+
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.event.EditorMouseEvent;
+import com.intellij.openapi.editor.event.EditorMouseEventArea;
+import com.intellij.openapi.editor.event.EditorMouseListener;
+import com.intellij.openapi.editor.ex.MarkupModelEx;
+import com.intellij.openapi.editor.ex.RangeHighlighterEx;
+import com.intellij.openapi.editor.markup.*;
+import com.intellij.openapi.fileEditor.TextEditor;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.openapi.wm.ToolWindow;
+import com.intellij.openapi.wm.ex.ToolWindowManagerEx;
+import com.intellij.ui.ColorUtil;
+import com.intellij.ui.JBColor;
+import icons.FlutterIcons;
+import io.flutter.run.daemon.FlutterApp;
+import io.flutter.utils.AnimatedIcon;
+import io.flutter.view.FlutterView;
+import io.flutter.view.InspectorPerfTab;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.awt.*;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+import java.util.List;
+
+class EditorPerfDecorations implements EditorMouseListener, EditorPerfModel {
```

Comment:
I'd add general dartdoc for the (3-4?) larger classes in this PR to help people understand the roles of the different classes.

Reply:
Added a bunch more top level comments.